.github/workflows/main.yml

```diff
@@ -167,32 +135,121 @@ jobs:
         run: |
           yum install -y java-1.8.0-openjdk-devel
 
-      - name: build zetasql
+      - name: build zetasql parser
         run: |
           source /opt/rh/devtoolset-8/enable
           source /opt/rh/rh-python38/enable
-          bazel build ${{ env.build_argv }} ${{ env.target }}
-
-      - name: Find Orphan Library
+          bazel build ${{ env.target }} ${{ env.build_argv }}
+      
+      - name: Test
         run: |
-          ./find_orphan_so.sh
+          bazel test ${{ env.build_argv }} --test_summary=detailed ${{ env.target }}
 
+      - name: build zetasql parser dependencies
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          bazel query 'deps(${{ env.target }})' |grep //zetasql|xargs bazel build ${{ env.build_argv }} 
+          bazel build "@com_googleapis_googleapis//:all" ${{ env.build_argv }}
+          bazel build "@com_google_file_based_test_driver//file_based_test_driver:all" ${{ env.build_argv }}
+          bazel build "@com_googlesource_code_re2//:re2" ${{ env.build_argv }}
+          
       - name: Determine Version
+        if: ${{ github.event_name == 'push' }}
         run: |
           VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
           VERSION=$(echo $VERSION | sed -e 's/^v//')
           echo "TAG=$VERSION" >> $GITHUB_ENV
+      
+      - name: pack libzetasql
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          ./pack_zetasql.sh
+        env:
+          TAG: ${{ env.TAG }}
 
+      - name: Upload Artifacts
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v2
+        with:
+          path: libzetasql-*.tar.gz
+          name: release-artifacts
+
+  macos-build:
+    runs-on: macos-latest
+    timeout-minutes: 120
+    env:
+      build_argv: '--features=-supports_dynamic_linker '
+      bazel_version: '3.7.2'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: update bazel version
+        run: echo ${{ env.bazel_version }} > .bazelversion
+
+      - name: install bazelisk
+        run: |
+          brew install bazelisk
+
+      - name: install utils tools
+        run: |
+          brew install coreutils
+          brew install binutils
+      - name: print toolchain info
+        run: |
+          gcc -v
+          g++ -v
+          python -V
+          java -version
+
+      - name: build
+        run: |
+          bazel build ${{ env.build_argv }} ${{ env.target }}
```

Comment:
we should create a compile script for easy reproduce on local machine

Reply:
Yep